### ISSUE TITLE
[BUG] Mage.ai doesn't work when sql server used for MAGE_DATABASE_CONNECTION_URL

### ISSUE THREAD

**Issue by tuokorTW** [mage-ai/mage-ai#4681]
### Mage version

v0.9.65

### Describe the bug

If I set MAGE_DATABASE_CONNECTION_URL to use sql server as per documentation here https://docs.mage.ai/guides/docker/connecting-a-database 
On startup mage will throw errors with stacktrace:
```
magic-1         | INFO:alembic.runtime.migration:Running upgrade dfe49d040487 -> 90d978a8aef8, Update unique constraint for secret
magic-1         | INFO:sqlalchemy.engine.Engine:ROLLBACK
magic-1         | Traceback (most recent call last):
magic-1         |   File "/usr/local/lib/python3.10/site-packages/mage_ai/server/scheduler_manager.py", line 40, in run_scheduler
magic-1         |     database_manager.run_migrations()
magic-1         |   File "/usr/local/lib/python3.10/site-packages/mage_ai/orchestration/db/database_manager.py", line 114, in run_migrations
magic-1         |     command.upgrade(alembic_cfg, revision='head')
magic-1         |   File "/usr/local/lib/python3.10/site-packages/alembic/command.py", line 403, in upgrade
magic-1         |     script.run_env()
magic-1         |   File "/usr/local/lib/python3.10/site-packages/alembic/script/base.py", line 583, in run_env
magic-1         |     util.load_python_file(self.dir, "env.py")
magic-1         |   File "/usr/local/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 95, in load_python_file
magic-1         |     module = load_module_py(module_id, path)
magic-1         |   File "/usr/local/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 113, in load_module_py
magic-1         |     spec.loader.exec_module(module)  # type: ignore
magic-1         |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
magic-1         |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
magic-1         |   File "/usr/local/lib/python3.10/site-packages/mage_ai/orchestration/db/migrations/env.py", line 103, in <module>
magic-1         |     run_migrations_online()
magic-1         |   File "/usr/local/lib/python3.10/site-packages/mage_ai/orchestration/db/migrations/env.py", line 97, in run_migrations_online
magic-1         |     context.run_migrations()
magic-1         |   File "<string>", line 8, in run_migrations
magic-1         |   File "/usr/local/lib/python3.10/site-packages/alembic/runtime/environment.py", line 948, in run_migrations
magic-1         |     self.get_context().run_migrations(**kw)
magic-1         |   File "/usr/local/lib/python3.10/site-packages/alembic/runtime/migration.py", line 627, in run_migrations
magic-1         |     step.migration_fn(**kw)
magic-1         |   File "/usr/local/lib/python3.10/site-packages/mage_ai/orchestration/db/migrations/versions/90d978a8aef8_update_unique_constraint_for_secret.py", line 25, in upgrade
magic-1         |     unique_constraints = insp.get_unique_constraints('secret')
magic-1         |   File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/reflection.py", line 631, in get_unique_constraints
magic-1         |     return self.dialect.get_unique_constraints(
magic-1         |   File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/interfaces.py", line 403, in get_unique_constraints
magic-1         |     raise NotImplementedError()
magic-1         | NotImplementedError
```
This seems to refer to this issue https://github.com/sqlalchemy/sqlalchemy/discussions/8317 which states that mssql has constraints as indexes. This probably needs a some kind of fix so that using mssql is possible as a mage database.

### To reproduce

1. clone for example mage-ai quickstart repo that has the docker-compose
2. Start a local sql server with like `docker run -e "ACCEPT_EULA=Y" -e "SA_PASSWORD=somesecret" -p 1433:1433 --name sql1 --hostname sql1 -d mcr.microsoft.com/mssql/server:2019-latest
` 
3. Create a database to the sql server for example named mage_ai `docker exec -it sql1 bash -c '/opt/mssql-tools/bin/sqlcmd -l 30 -S localhost -U sa -P "somesecret" -Q "CREATE DATABASE mage_ai"'`
3. Make sure .env file contains line: `MAGE_DATABASE_CONNECTION_URL="mssql+pyodbc://?odbc_connect=DRIVER={ODBC Driver 18 for SQL Server};SERVER=host.docker.internal;DATABASE=mage_ai;UID=sa;PWD=somesecret;ENCRYPT=yes;TrustServerCertificate=yes;"`
4. run `docker-compose up` in the mage-ai quickstart to start the mage.ai container
5. Observe logs containing errors related to sqlalchemy.

### Expected behavior

Database table creations should complete and mage.ai should be usable with sql server as the database.

**Comment by tuokorTW** [comment]
Looks like the problem line is this https://github.com/mage-ai/mage-ai/blame/04c8a08e60e4726234bd71f6e7ef87f002278310/mage_ai/orchestration/db/migrations/versions/90d978a8aef8_update_unique_constraint_for_secret.py#L25

**Comment by johnson-mage** [comment]
> Looks like the problem line is this https://github.com/mage-ai/mage-ai/blame/04c8a08e60e4726234bd71f6e7ef87f002278310/mage_ai/orchestration/db/migrations/versions/90d978a8aef8_update_unique_constraint_for_secret.py#L25

@dy46 There might be an issue related to a db migration for secrets.

**Comment by dy46** [comment]
we don't officially support MSSQL as a database for mage right now. are you able to use sqlite or postgres?

**Comment by dy46** [comment]
We've never fully tested using Mage with MSSQL, so I can't guarantee that all features would work as expected

**Comment by tuokorTW** [comment]
Ok, maybe then I don't yet use mssql in production for now. I'll do a quick test locally if I can fix this issue so that at least migrations would block mssql usage in the future.

**Comment by wangxiaoyou1993** [comment]
The fix is included in version 0.9.66.
